FirebaseUI/Database/Implementation/FirebaseArray.m

```diff
@@ -20,113 +20,142 @@
 
 #import "FirebaseArray.h"
 
+@interface FirebaseArray ()
+
+/**
+ * The delegate object that array changes are surfaced to.
+ */
+@property(strong, nonatomic) NSMutableArray<FIRDataSnapshot *> * snapshots;
+
+/**
+ * A set containing the query observer handles that should be released when
+ * this array is freed.
+ */
+@property(strong, nonatomic) NSMutableSet<NSNumber *> *handles;
+
+@end
+
 @import FirebaseDatabase;
 
 @implementation FirebaseArray
 
-#pragma mark -
-#pragma mark Initializer methods
-
-- (instancetype)initWithRef:(FIRDatabaseReference *)ref {
-  return [self initWithQuery:ref];
-}
+#pragma mark - Initializer methods
 
 - (instancetype)initWithQuery:(FIRDatabaseQuery *)query {
+  NSParameterAssert(query != nil);
```

Comment:
This worries me slightly - would the behavior in the previous version have been a crash if a null query was passed anyway?